File: spring-data-requery/src/main/java/org/springframework/data/requery/converters/OffsetDateTimeToStringConverter.java
package org.springframework.data.requery.converters;

import io.requery.Converter;
import org.springframework.util.StringUtils;

import javax.annotation.Nullable;
import java.time.OffsetDateTime;

/**
 * org.springframework.data.requery.converters.OffsetDateTimeToStringConverter
 *
 * @author debop
 */
public class OffsetDateTimeToStringConverter implements Converter<OffsetDateTime, String> {

    @Override
    public Class<OffsetDateTime> getMappedType() {
        return OffsetDateTime.class;
    }

    @Override
    public Class<String> getPersistedType() {
        return String.class;
    }

    @Nullable
    @Override
    public Integer getPersistedSize() {
        return null;
    }

    @Override
    public String convertToPersisted(OffsetDateTime value) {
        return (value != null)
               ? Java8Times.toIsoOffsetDateTimeString(value)
               : null;
    }

    @Override
    public OffsetDateTime convertToMapped(Class<? extends OffsetDateTime> type, @Nullable String value) {
        return (StringUtils.hasText(value))
               ? OffsetDateTime.parse(value)
               : null;
    }
}

File: spring-data-requery/src/main/kotlin/org/springframework/data/requery/converters/OffsetDateTimeToStringConverter.kt
package org.springframework.data.requery.converters

import io.requery.Converter
import java.time.OffsetDateTime

/**
 * Convert [OffsetDateTime] to [String]
 * @author debop
 * @since 18. 7. 2
 */
class OffsetDateTimeToStringConverter: Converter<OffsetDateTime, String> {

    override fun getPersistedType(): Class<String> = String::class.java
    override fun getMappedType(): Class<OffsetDateTime> = OffsetDateTime::class.java
    override fun getPersistedSize(): Int? = null

    override fun convertToMapped(type: Class<out OffsetDateTime>?, value: String?): OffsetDateTime? {
        return value?.let { OffsetDateTime.parse(it) }
    }

    override fun convertToPersisted(value: OffsetDateTime?): String? {
        return value?.toIsoOffsetDateTimeString()
    }
}